Blog post on Astro 1.4.0 (#319)
withastro/astro.build: src/content/blog/astro-140.mdx

---
title: 'Astro 1.4.0 Release'

> description: 'Introducing: Responsive partial hydration • Solid.js support • Lit SSR support • Named slots • Global style support • and more!'

Author: woops, update

Making this switch we didn't realize that the ordering of styles was ideal. In 1.4.0 styles you write in Astro components will come *after* global styles imported. In practice this means that you write a component like so:

__Title.astro__

```astro
---
import '../styles/global-titles.css';
---
<h1>{ Astro.props.title }</h1>
<style>
  h1 {
    color: aliceblue;
  }
</style>
```

When your site is built and the CSS is bundled you get something like the following:

```css
/* /src/styles/global-titles.css */
h1 {
  color: onyx;
}

/* /src/components/Title.astro */
h1:where(.astro-1234) {
  color: aliceblue;
}
```

Given that these selectors both have the same specificity, the one lowest in order, the one you defined in your Title.astro component will prevail within that component. Any `h1` defined outside of your component will be `onyx` color.

You get the best of both worlds; styles defined in Astro are prioritized given the same specificity, but you can override them by increasing specificity when needed.

The order of CSS in Astro now is:

1. CSS in your `public/` folder that add as `<link>` tags in the head.

> 1. CSS you import in your frontmatter, ordered by the order of your import statements (depth-first).

Reviewer: ```suggestion
1. CSS imported in your frontmatter, in the order of your import statements (depth-first).
```
What do we mean by depth-first?

1. CSS defined in Astro style tags.

> We're currently writing up full document to explain how CSS ordering works and how you can control CSS priority via ordering and specificity.

Reviewer: ```suggestion
We're currently writing up full documentation to explain how CSS ordering works and how you can control CSS priority via ordering and specificity.
```

## JSX support in Vue

<BlogContentImage src={vueJsx} alt="Code example that shows a JSX component defined with Vue" />

> Using the [Vue integration](https://github.com/withastro/astro/blob/d08093f7d4f9762c0edc71e14c796d7ea8038cda/packages/integrations/vue/README.md) you can now use JSX and TSX. Setting `jsx: true` in the integrations options enables the [@vitejs/plugin-vue-jsx](https://github.com/vitejs/vite/tree/main/packages/plugin-vue-jsx) Vite plugin which does most of the work.

Reviewer: ```suggestion
You can now use JSX and TSX with the [Vue integration](https://github.com/withastro/astro/blob/d08093f7d4f9762c0edc71e14c796d7ea8038cda/packages/integrations/vue/README.md). Setting `jsx: true` in the integrations options enables the [@vitejs/plugin-vue-jsx](https://github.com/vitejs/vite/tree/main/packages/plugin-vue-jsx) Vite plugin, which does most of the work, and you can use any of its options.
```
